src/buffer/out/AttrRow.cpp

```diff
@@ -288,21 +288,40 @@ void ATTR_ROW::ReplaceAttrs(const TextAttribute& toBeReplacedAttr, const TextAtt
         // Check for that circumstance by seeing if we're inserting a single run of the
         // left side color right at the boundary and just adjust the counts in the existing
         // two elements in our internal list.
-        else if (_list.size() == 2 && newAttrs.at(0).GetLength() == 1)
+        else if (iStart == iEnd && iStart > 0)
         {
-            const auto left = _list.begin();
-            if (iStart == left->GetLength() && NewAttr == left->GetAttributes())
+            size_t lowerBound = 0;
+            size_t upperBound = 0;
+            for (int i = 0; i < _list.size(); i++)

```

Comment:
`std::list::size() `return `size_t`, so this should be:

`for (size_t i = 0; i < _list.size(); i++)`